File: app/src/main/java/tk/lorddarthart/rxnewstestapp/PagerAdapter.java
package tk.lorddarthart.rxnewstestapp;

import android.os.Parcelable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.app.FragmentStatePagerAdapter;

public class PagerAdapter extends FragmentPagerAdapter {
    private int mNumOfTabs;

    PagerAdapter(FragmentManager fm, int NumOfTabs) {
        super(fm);
        this.mNumOfTabs = NumOfTabs;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new MainMainFragment();
            case 1:
                return new MainVipFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return mNumOfTabs;
    }

    @Override
    public void restoreState(Parcelable state, ClassLoader loader) {
        super.restoreState(state, loader);
    }
}

File: app/src/main/java/tk/lorddarthart/rxnewstestapp/app/view/adapter/PagerAdapter.kt
package tk.lorddarthart.rxnewstestapp.app.view.adapter

import android.os.Parcelable

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import tk.lorddarthart.rxnewstestapp.app.view.fragment.general.main.pages.main.MainPage
import tk.lorddarthart.rxnewstestapp.app.view.fragment.general.main.pages.vip.VipPage

class PagerAdapter internal constructor(fm: FragmentManager, private val mNumOfTabs: Int) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return MainPage()
            1 -> return VipPage()
            else -> return MainPage()
        }
    }

    override fun getCount(): Int {
        return mNumOfTabs
    }

    override fun restoreState(state: Parcelable?, loader: ClassLoader?) {
        super.restoreState(state, loader)
    }
}
